v3/src/components/graph/models/data-configuration-model.ts

```diff
@@ -443,9 +443,9 @@
           primaryRole = self.primaryRole
         return primaryRole === role || !['left', 'bottom'].includes(place)
       },
-      graphPlaceCanAcceptAttributeIDDrop(place: GraphPlace, idToDrop?: string) {
+      graphPlaceCanAcceptAttributeIDDrop(place: GraphPlace, dataSet?: IDataSet, idToDrop?: string) {
```

Comment:
If the incoming dataset is different than the current dataset, the logic will be quite different. For example, it doesn't make sense to drop an attribute from a different dataset on the top or rightCat axes. On the other hand we don't want to forbid dropping an attribute from a different dataset on x or y even though the consequences will be to remove current attributes. Bottomline: we need to test whether the incoming dataset is the same as the current one and, if not, allow drops only on 'x' and 'y'.

Reply:
Good point -- I'll add a commit to address this.